### PR TITLE
Add ProjectCentralUtils utility

### LINKED ISSUE #31200: Provide a util API to pull a package from central
**Description:**
Prerequisite for #30757

### PR DESCRIPTION
**Pull request by Ibaqu**
## Purpose
Add class and resolve dependencies for pulling packages from central

Fixes #31200

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectCentralUtils.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
+package io.ballerina.projects.util;
```
Comment (IrushiL) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectCentralUtils.java:1]: Shall we add the license header?

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectCentralUtils.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            TomlDocument settingsTomlDocument = TomlDocument
+                    .from(String.valueOf(settingsFilePath.getFileName()), Files.readString(settingsFilePath));
+            SettingsBuilder settingsBuilder = SettingsBuilder.from(settingsTomlDocument);
+            return settingsBuilder.settings();
+        } catch (IOException e) {
+            // If Settings.toml not exists return empty Settings object
+            return Settings.from();
+        }
+    }
+
```
Comment (IrushiL) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectCentralUtils.java:83]: Extra new line.

Comment (Ibaqu) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectCentralUtils.java]: Removed new line

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectCentralUtils.java (hunk 1 above)
Comment (Ibaqu) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectCentralUtils.java:19]: Added license header for all files

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PullCommand.java [hunk 3]
```diff
@@ -175,28 +168,14 @@ public void execute() {
...
-                                   RepoUtils.getBallerinaVersion(), false);
-            } catch (PackageAlreadyExistsException e) {
-                errStream.println(e.getMessage());
-                CommandUtil.exitError(this.exitWhenFinish);
-            } catch (CentralClientException e) {
-                errStream.println("unexpected error occurred while pulling package:" + e.getMessage());
-                CommandUtil.exitError(this.exitWhenFinish);
-            }
+        try {
+            ProjectCentralUtils.pullPackage(orgName, packageName, version);
```
Comment (pramodya1994) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PullCommand.java:172]: If we change `pullPackage` implememntation, we should check all the scenarios of `pull` command.

**Comment by Ibaqu** [comment]
Implementation has changed which will be handled at the LS side

**Pull request closed without merging**